modules/configuring-ovnk-use-second-ovs-bridge.adoc

```diff
@@ -0,0 +1,295 @@
++
+[NOTE]
+====
+The resource lists the name of the interface that `br-ex` selects for your cluster. By default, `br-ex` selects the primary interface for a node based on boot order and the IP address subnet in the machine network. Certain machine network configurations might require that `br-ex` continues to select the default interfaces or bonds for a host node.
+====
++
+.. Create a machine configuration file on the host node to override the default interface.
++
+[IMPORTANT]
+====
+Only create this machine configuration file for the purposes of changing the `br-ex` selection logic. Using this file to change the IP addresses of existing nodes in your cluster is not supported.
+====
++
+.Example of a machine configuration file that overrides the default interface
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 12-worker-br-ex-override
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+        - path: /var/lib/ovnk/iface_default_hint
+          mode: 0420
+          overwrite: true
+          contents:
+            source: data:text/plain;charset=utf-8,bond0
+          filesystem: root
+----
++
+.. Before you apply the configuration to all new nodes in your cluster, reboot the host node to verify that `br-ex` selects the intended interface and does not conflict with the new interfaces that you defined on `br-ex1`.
++
+.. Apply the machine configuration file to all new nodes in your cluster:
++
+[source,terminal]
+----
+$ oc create -f <machine_config_file_name>
+----
+
+.Verification
+
+. Identify the IP addresses of nodes with the `exgw-ip-addresses` label in your cluster to verify that the nodes use the additional bridge instead of the default bridge:
++
+[source,terminal]
+----
+$ oc get nodes -o json | grep --color exgw-ip-addresses
+----
++
+.Example output
+[source,terminal]
+----
+"k8s.ovn.org/l3-gateway-config":
+   \"exgw-ip-address\":\"172.xx.xx.yy/24\",\"next-hops\":[\"xx.xx.xx.xx\"],
+----
+
+. Observe that the additional bridge exists on target nodes by reviewing the network interface names on the host node:
++
+[source,terminal]
+----
+$ oc debug node/<node_name> -- chroot /host sh -c "ip a | grep mtu | grep br-ex"
+----
++
+.Example output
+[source,terminal]
+----
+Starting pod/worker-1-debug ...
+To use host binaries, run `chroot /host`
+# ...
+5: br-ex: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
+6: br-ex1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default qlen 1000
+----
+
+. Optional: If you use `/var/lib/ovnk/iface_default_hint`, check that the MAC address of `br-ex` matches the MAC address of the primary selected interface:
++
+[source,terminal]
+----
+$ oc debug node/<node-name> -- chroot /host sh -c "ip a | grep -A1 -E 'br-ex|ens3'
```

Comment:
above is using `bond0`, here if we need to check `br-ex|bond0`?

Reply:
:heavy_check_mark: